src/oic/oauth2/__init__.py

```diff
@@ -181,6 +182,7 @@ def __init__(
         timeout=None,
         message_factory: Type[MessageFactory] = OauthMessageFactory,
         settings: PyoidcSettings = None,
+        request_session: Session = None
```

Comment:
I would rather include this in the `settings` class. The number of `kwargs`  is already quite high...